### PR TITLE
docker: Add an architecture overview

### PR DESCRIPTION
**Pull request by doanac**
Signed-off-by: Andy Doan <andy@foundries.io>

### CODE REVIEW

**Comment by doanac** [comment]
I'm not sure if I'm providing enough here or if its at the right level.

**Comment by doanac** [comment]
Docs for 1541974f19117267ffd887f07538a427c6525963 are browesable at: https://ci.foundries.io/projects/fio-docs/builds/677/docs/artifacts/html/index.html

File: source/reference-manual/docker/docker-architecture.rst [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      }
+    ...
+
+This Target includes 2 compose apps, ``fiotest`` and ``shellhttpd`` that
+will come from a Factory's containers.git repository. The apps have
+a "pinned" URL that includes the sha256 checksum of its content.
+
+The container build process will pin each container defined in the
+compose file to an immutable sha256 Docker reference to make sure
+the compose definition can't change over time. The app is then
```
Comment (EmbeddedAndroid) [inline source/reference-manual/docker/docker-architecture.rst:35]: We might want to add a sentence here to explain if you use a container image  not built by the factory, that you will not get a immutable sha256 docker reference.

**Comment by EmbeddedAndroid** [comment]
One minor comment about using external containers, but otherwise looks great.

File: source/reference-manual/docker/docker-architecture.rst (hunk 1 above)
Comment (doanac) [reply source/reference-manual/docker/docker-architecture.rst:35]: we actually pin those containers as well.

**Pull request merged**